src/types.ts

```diff
@@ -1669,9 +1669,17 @@ export class ZodObject<
 
     const { shape, keys: shapeKeys } = this._getCached();
     const extraKeys: string[] = [];
-    for (const key in ctx.data) {
-      if (!shapeKeys.includes(key)) {
-        extraKeys.push(key);
+
+    if (
+      !(
```

Comment:
Couldn't this `!(A && B)` be simplified to `!B || !A` so it can short-circuit?

I'd also guess that putting B first is faster as it's a simple comparison, but not sure on that